packages/instrumentation-qdrant/src/instrumentation.ts

```diff
@@ -0,0 +1,347 @@
+          () => {
+            return context.with(execContext, () => {
+              return original.apply(this, args);
+            });
+          },
+          (e) => {
+            if (e) {
+              plugin._diag.error(`Error in Qdrant instrumentation`, e);
+            }
+          },
+        );
+        const wrappedPromise = execPromise
+          .then((result: any) => {
+            return new Promise((resolve) => {
+              plugin._endSpan({ methodName: original.name, span, result });
+              resolve(result);
+            });
+          })
+          .catch((error: Error) => {
+            return new Promise((_, reject) => {
+              span.setStatus({
+                code: SpanStatusCode.ERROR,
+                message: error.message,
+              });
+              span.end();
+              reject(error);
+            });
+          });
+        return context.bind(execContext, wrappedPromise as any);
+      };
+    };
+  }
+  private _startSpan({
+    collectionName,
+    params,
+    methodName,
+  }: {
+    collectionName: string;
+    params: UpsertRequest | DeleteRequest | RetrieveRequest | SearchRequest;
+    methodName: string;
+  }): Span {
+    const spanName = `qdrant.${methodName}`;
+    const span = this.tracer.startSpan(spanName, {
+      kind: SpanKind.CLIENT,
+    });
+    span.setAttribute(SpanAttributes.VECTOR_DB_VENDOR, "Qdrant");
+
+    try {
+      if (this._config.traceContent) {
+        if (methodName === UPSERT) {
+          this._setUpsertAttributes(
+            span,
+            collectionName,
+            params as UpsertRequest,
+          );
+        } else if (methodName === DELETE) {
+          this._setDeleteAttributes(
+            span,
+            collectionName,
+            params as DeleteRequest,
+          );
+        } else if (methodName === RETRIEVE) {
+          this._setRetrieveAttributes(
+            span,
+            collectionName,
+            params as RetrieveRequest,
+          );
+        } else if (methodName === SEARCH) {
+          this._setSearchAttributes(
+            span,
+            collectionName,
+            params as SearchRequest,
+          );
+        }
+      }
+    } catch (e) {
+      this._diag.warn(e);
+      this._config.exceptionLogger?.(e);
+    }
+
+    return span;
+  }
+
+  private _endSpan({
+    methodName,
+    span,
+    result,
+  }: {
+    methodName: string;
+    span: Span;
+    result: any;
+  }) {
+    try {
+      if (methodName === SEARCH) {
+        this._setSearchResultAttributes(span, result as SearchResponse);
+      }
+      span.setStatus({ code: SpanStatusCode.OK });
+      span.end();
+    } catch (e) {
+      this._diag.debug(e);
+      this._config.exceptionLogger?.(e);
+    }
+  }
+
+  private _setUpsertAttributes(
+    span: Span,
+    collectionName: string,
+    params: qdrant.Schemas["PointInsertOperations"],
+  ) {
+    span.setAttribute("db.qdrant.upsert.collection_name", collectionName);
```

Comment:
Can we put these constants in the semantic conventions package and reference them here?

Reply:
Do you mean all the literals or the common ones?

Reply:
All the literals. Other systems then take these and automatically build dashboards from them

Reply:
[`886077d` (#278)](https://github.com/traceloop/openllmetry-js/pull/278/commits/886077dac4a49aa3859374d203fdd6809238fce2)